pkg/segment/segment.go

```diff
@@ -215,11 +216,12 @@ func GetUserIdentity(telemetryFilePath string) (string, error) {
 	}
 
 	// check if the id is a valid uuid, if not, nil is returned
-	if uuid.Parse(strings.TrimSpace(string(id))) == nil {
-		id = []byte(uuid.NewRandom().String())
-		if err := os.WriteFile(telemetryFilePath, id, 0600); err != nil {
+	if _, err := uuid.ParseBytes(bytes.TrimSpace(id)); err != nil {
+		uid := uuid.New()
```

Comment:
`uuid.New()` would panic if it couldn't return an UUID. Could you please use `uuid.NewRandom()` instead and handle the error instead?

Reply:
Fixed in https://github.com/redhat-developer/odo/pull/7173/commits/e4bb039e7a7d89b58f9d12865ec164cf49d571cf.